server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -90,9 +90,21 @@ public static class Defaults {
      */
     @PublicApi(since = "1.0.0")
     public enum Dynamic {
-        TRUE,
-        FALSE,
-        STRICT
+        TRUE("true"),
+        FALSE("false"),
+        STRICT("strict"),
+        STRICT_ALLOW_TEMPLATES("strict_allow_templates");
+
+        private final String name;
+
+        Dynamic(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public String toString() {
+            return this.name;
```

Comment:
You could just use `name().toLowerCase();` and drop the `name` field:

```suggestion
        @Override
        public String toString() {
            return name().toLowerCase();
        }
```

Reply:
Will that cause performance issue if we call `toLowerCase()` frequently?

Reply:
We call it only when exception are raised, right? We probably could replace the `dynamic.toString()` with `dynamic.name().toLowerCase()` in such cases (to eliminate any accidental `toString` conversions), what do you think?

Reply:
That's better, I've changed the code, thank you!